File: CS502018/pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    //checks for proper arguements
    if (argc != 2 || atoi(argv[1]) < 0)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }

    //set key to argv[1] and wrap it around
    int key = atoi(argv[1]);
    while(key > 26)
    {
        key = (key % 26);
    }

    //ask user for plaintext
    string plainText = get_string("plaintext: ");

    //gets the plaintext's length
    int plainLen = strlen(plainText);

    //prints ciphertext
    printf("ciphertext: ");

    //goes through each element in the plaintext
    for(int i = 0; i < plainLen; i++)
    {
        //checks if if the plaintext is alphabetical
        if(isalpha(plainText[i]))
        {
            //checks if letter is uppercase
            if(isupper(plainText[i]))
            {
                //adds key to the letter
                int cWord = (((int)plainText[i]) + key);

                //wraps around if needed
                if(cWord > 90)
                {
                    cWord = (((int)cWord) % 91) + 65;
                }

                //printf the cipher letter
                printf("%c",cWord);
            }

            //checks if letter is lower case
            else if (islower(plainText[i]))
            {
                //adds key to letter
                int cWord = (((int)plainText[i]) + key);

                //wraps around if needed
                if(cWord > 122)
                {
                    cWord = (((int)cWord) % 123) + 97;
                }

                //prints cipher letter
                printf("%c",cWord);
            }
        }

        //if letter is not alphabetical print character
        else
            printf("%c", plainText[i]);
    }

//adds new line
printf("\n");

return 0;
}

File: CS502018/pset6/sentimental/caesar/caesar.py
import sys
import cs50

def main():

    #check if proper arguements are passed
    if len(sys.argv) != 2:
        print("Usage: python caesar.py k")
        sys.exit(1)

    #cast key from string to int
    key = int(sys.argv[1])

    #get the plaintext
    plaintext = cs50.get_string("plaintext: ")

    #run the plaintext into the cipher function
    ciphertext = cipher(plaintext, key)

    #prints the cipher text
    print(f"ciphertext: {ciphertext}")

def cipher(plaintext, key):

    #declare ciphertext string
    ciphertext = ""

    #go through each character and cipher it
    for i in range(len(plaintext)):

        #if the letter is upper
        if plaintext[i].isupper():

            #runs the algorithm
            ciphertext += chr(((ord(plaintext[i]) - 65 + key) % 26) + 65)

        #if the letter is lower
        elif plaintext[i].islower():

            #runs the algorithm
            ciphertext += chr(((ord(plaintext[i]) - 97 + key) % 26) + 97)

        #if it is a special character
        else:

            #just add the character to the string
            ciphertext += plaintext[i]

    #return ciphertext
    return ciphertext

if __name__ == "__main__":
    main()